Src/IronPython/Runtime/Types/PythonTypeUserDescriptorSlot.cs

```diff
@@ -74,17 +73,11 @@ public sealed class PythonTypeUserDescriptorSlot : PythonTypeSlot {
         }
 
         internal override bool IsSetDescriptor(CodeContext context, PythonType owner) {
-            object dummy;
-            return PythonOps.TryGetBoundAttr(context, Value, "__set__", out dummy);
+            PythonType pt = DynamicHelpers.GetPythonType(Value);
+            if (pt.IsSystemType) return false;
+            return pt.TryResolveSlot(context, "__set__", out _);
```

Comment:
I haven't looked at this for time, but why the switch from TryGetBoundAttr to TryResolveSlot?